### PR TITLE
RFC 6265bis: Add Lax-allowing-unsafe SameSite enforcement mode

### PR DESCRIPTION
**Pull request by chlily1**
Addresses #1422

### CODE REVIEW

**Comment by chlily1** [comment]
@englehardt PTAL.

Ended up adapting the text from Cookie Incrementalism into two sections, one with a focus on the behavior (with a monkey patch to the algorithm) and the other more focusing on the justification for it.

File: draft-ietf-httpbis-rfc6265bis.md [hunk 1]
```diff
@@ -1303,6 +1303,45 @@ When possible, developers should use a session management mechanism such as
...
+`SameSite` enforcement mode, may be set for top-level navigations, regardless of
+HTTP request method, as specified in {{storage-model}}.)
+
+"Lax-allowing-unsafe" is not a distinct value of the `SameSite` attribute.
+Rather, user agents MAY apply "Lax-allowing-unsafe" enforcement only to cookies
+that did not explicitly specify a `SameSite` attribute (i.e., those whose
+same-site-flag was set to "Default" by default). If a user agent applies
+"Lax-allowing-unsafe" enforcement, the user agent SHOULD restrict
+"Lax-allowing-unsafe" enforcement to cookies whose age is less than a duration
+of the user agent’s choosing (2 minutes seems reasonable).
```
Comment (englehardt) [inline draft-ietf-httpbis-rfc6265bis.md:1324]: I think the wording here is a bit open ended. How about:

"To limit the scope of this compatibility mode, user agents which apply "Lax-allowing-unsafe" SHOULD restrict the enforcement to cookies which were created recently. Deployment experience has shown that a cookie age of 2 minutes or less to be a reasonable limit."

**Review by englehardt — APPROVED** [review]
Overall this LGTM. Having this as a monkey patch seems like the best solution. I've added two suggestions for alternate wording.

File: draft-ietf-httpbis-rfc6265bis.md [hunk 2]
```diff
@@ -2041,6 +2080,33 @@ reload navigation triggered through the user interface may replay the original
...
 cross-site, like the request that initially navigated to the page.
 
+### Top-level requests with "unsafe" methods {#unsafe-top-level-requests}
+
+The "Lax" enforcement mode described in {{strict-lax}} allows a cookie to be
+sent with a cross-site HTTP request if and only if it is a top-level navigation
+with a "safe" HTTP method. Implementation experience shows that this is
+difficult to apply as the default behavior, as some sites may rely on cookies
+not explicitly specifying a `SameSite` attribute being included on top-level
+cross-site requests with "unsafe" HTTP methods (as was the case in a previous
```
[the remark this answers is not in the document]
Comment (chlily1) [reply draft-ietf-httpbis-rfc6265bis.md]: Done.

File: draft-ietf-httpbis-rfc6265bis.md (hunk 1 above)
Comment (chlily1) [reply draft-ietf-httpbis-rfc6265bis.md]: Done.

**Comment by chlily1** [comment]
Thanks for the review!
This is dependent on 6265-non-http-apis so I'll merge it after that one lands.

**Pull request merged**